### ISSUE TITLE
Show only the source domain in article lists and article pages

### ISSUE THREAD

**Issue by quamsta** [uiowa/uiowa#7034]
# Story

As a webmaster, I'd like to see a truncated version of the "Source" link in articles, since long, bare URLs present accessibility issues because screen readers will often read a full URL letter by letter (source: [nysed.gov](https://www.nysed.gov/webaccess/create-accessible-hypertext-links))

> Hypertext URLs (such as https://www.engageny.org/resource/guidance-on-new-york-s-annual-professional-performance-review-law-and-regulations) should be avoided. Screen readers will read the URL letter by letter which would be very difficult to interpret for a user. Short URLs may be used occasionally if the point is to convey what the URL is to the user, such as for website homepages such as http://www.nysed.gov/ or https://portal.nysed.gov/. However, if this is not necessary, then it's better to display links as readable text such as the [New York State Education Department](http://www.nysed.gov/) website.

# Proposed solution

Strip out everything except the top level domain from the source when displaying the source link, but still link to the original source. 

### Before:
<img width="1135" alt="Screenshot 2023-10-24 at 8 27 40 AM" src="https://github.com/uiowa/uiowa/assets/472923/1e302121-37eb-4d1d-8a7a-3816cb78d13f">

### After:
<img width="1102" alt="Screenshot 2023-10-24 at 8 28 06 AM" src="https://github.com/uiowa/uiowa/assets/472923/c854eda7-26dc-4003-b63b-0e1e2492b3fe">

### Before:
<img width="1326" alt="Screenshot 2023-10-24 at 8 25 13 AM" src="https://github.com/uiowa/uiowa/assets/472923/8de9a08d-94d1-4494-adbd-698c97d1b1c3">

### After:
<img width="1373" alt="Screenshot 2023-10-24 at 8 25 30 AM" src="https://github.com/uiowa/uiowa/assets/472923/1c4cfac5-0280-4cf3-8390-e3d40400424f">

**Comment by bspeare** [comment]
@quamsta what about adding a required "Source link title" so that the link title can be used instead?  That seems to be what your [source](https://www.nysed.gov/webaccess/create-accessible-hypertext-links) and https://webaim.org/techniques/hypertext/link_text#urls suggest.

**Comment by quamsta** [comment]
@bspeare I think that's better than a full URL, but imho a link title doesn't really describe the source of the information, but would instead likely repeat parts (if not all) of the article title already shown above unless it included the source domain name in parenthesis or something.

Example:
[Stephanie Peterson successfully defends and publishes her Ph.D. Thesis (medicine.uiowa.edu)](https://medicine.uiowa.edu/pathology/content/stephanie-peterson-successfully-defends-and-publishes-her-phd-thesis)

The way that I understand how screen readers work is that they will read a full URL letter by letter if it begins with "http(s)://", but would generally read a domain (without http(s)://) as a word. I confirmed this with Apple's "Spoken Content" accessibility option, so it's true for that at least.

I think I'd prefer to keep at least part of the source domain because that way we're succinctly saying that the source is "medicine.uiowa.edu" (the website) but a full title would be okay as long as it still included the domain since that's the source.
